src/high-level/high-level.lisp

```diff
@@ -596,39 +596,42 @@ it must be that KA = KB, and the resulting matrix is M x N."
       (magicl.lapack-cffi::%zungqr m n k a lda tau work lwork info)
       (make-matrix :rows m :cols n :data a))))
 
-(defun svd (m)
-  "Find the SVD of a matrix M. Return (VALUES U SIGMA Vt) where M = U*SIGMA*Vt"
-  (let ((jobu "A")
-        (jobvt "A")
-        (rows (matrix-rows m))
-        (cols (matrix-cols m))
-        (a (copy-matrix-storage (matrix-data m)))
-        (lwork -1)
-        (info 0))
-    (let ((lda rows)
-          (s (make-D-storage (min rows cols)))
-          (ldu rows)
-          (ldvt cols)
-          (work1 (make-Z-storage (max 1 lwork)))
-          (work nil)
-          (rwork (make-D-storage (* 5 (min rows cols)))))
-      (let ((u (make-Z-storage (* ldu rows)))
-            (vt (make-Z-storage (* ldvt cols))))
-        ;; run it once as a workspace query
-        (magicl.lapack-cffi::%zgesvd jobu jobvt rows cols a lda s u ldu vt ldvt
-                                     work1 lwork rwork info)
-        (setf lwork (round (realpart (aref work1 0))))
-        (setf work (make-Z-storage (max 1 lwork)))
-        ;; run it again with optimal workspace size
-        (magicl.lapack-cffi::%zgesvd jobu jobvt rows cols a lda s u ldu vt ldvt
-                                     work lwork rwork info)
-        (let ((smat (make-D-storage (* rows cols))))
-          (dotimes (i (min rows cols))
-            (setf (aref smat (column-major-index rows i i))
-                  (aref s i)))
-          (values (make-matrix :rows rows :cols rows :data u)
-                  (make-matrix :rows rows :cols cols :data smat)
-                  (make-matrix :rows cols :cols cols :data vt)))))))
+(defun svd (m &key reduced)
+  "Find the SVD of a matrix M. Return (VALUES U SIGMA Vt) where M = U*SIGMA*Vt. If REDUCED is non-NIL, return the reduced SVD (where either U or V are just partial isometries and not necessarily unitary matrices)."
+  (let* ((jobu (if reduced "S" "A"))
+         (jobvt (if reduced "S" "A"))
```

Comment:
@appleby @stylewarning  do you see what i see

Reply:
if you squint that `v` almost looks like `ν`